### PR TITLE
Create PillLogger class and update logger usage

### PR DESCRIPTION
**Pull request by cnivedit**
Create PillLogger class with file and console handlers
Update logger instantiation for existing usage

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,4 +15,10 @@ bin/
 
 /text-ui-test/ACTUAL.TXT
 text-ui-test/EXPECTED-UNIX.TXT
+
+# Local data and log files
 /data/
+/log/
+
+# Artifact manifest file
+/META-INF/
```
Comment (yakultbottle) [inline .gitignore:24]: Good job updating the gitignore, I definitely neglected this

File: src/main/java/seedu/pill/Pill.java [hunk 2]
```diff
@@ -5,34 +5,33 @@
...
 
-import java.util.logging.Level;
 import java.util.logging.Logger;
 
 public final class Pill {
     private static final Storage storage = new Storage();
     private static ItemMap items = new ItemMap();
     private static final Ui ui = new Ui(items);
     private static Parser parser = new Parser(items, storage);
+    private static final Logger logger = PillLogger.getLogger();
```
Comment (yakultbottle) [inline src/main/java/seedu/pill/Pill.java:19]: I like the usage of the private static final modifiers, enforcing OOP principles

File: src/main/java/seedu/pill/util/PillLogger.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.logging.ConsoleHandler;
+import java.util.logging.Level;
+import java.util.logging.FileHandler;
+import java.util.logging.SimpleFormatter;
+
+/**
+ * A utility class that manages a logger for the application, logging to both
+ * console and file. Logs are stored in {@code PillLog.log} under the {@code ./log/} directory.
+ * Console logging is disabled ({@code Level.OFF}), while file logging captures all levels.
+ */
```
Comment (yakultbottle) [inline src/main/java/seedu/pill/util/PillLogger.java:16]: Good job adding javadoc comments to make code easier to maintain

File: src/main/java/seedu/pill/util/PillLogger.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        try {
+            File dir = new File(PATH);
+            if (!dir.exists()) {
+                dir.mkdirs();
+            }
+            Handler fileHandler = new FileHandler(PATH + FILE_NAME, true);
+            fileHandler.setFormatter(new SimpleFormatter());
+            fileHandler.setLevel(Level.ALL);
+            logger.addHandler(fileHandler);
+        } catch (IOException e) {
```
Comment (yakultbottle) [inline src/main/java/seedu/pill/util/PillLogger.java:42]: Could this have extra empty lines to make different sections of the code readable?

**Review by yakultbottle — APPROVED** [review]
LGTM

File: src/main/java/seedu/pill/util/PillLogger.java (hunk 4 above)
Comment (cnivedit) [reply src/main/java/seedu/pill/util/PillLogger.java]: Improve line spacing in commit 242f484

**Review by yakultbottle — APPROVED** [review]

**Pull request merged**